lib/errors/error-collector.js

```diff
@@ -270,18 +284,6 @@ class ErrorCollector {
       return false
     }
 
-    // allow enabling & disabling the error tracer at runtime
-    // TODO: it would be better to check config in the public add() to prevents collecting
-    // errors on the transaction unnecessarily. Should we allow the work above or
-    // short-circuit sooner?
-    if (
-      !this.config.collect_errors ||
-      !this.config.error_collector ||
-      !this.config.error_collector.enabled
-    ) {
-      return false
-    }
-
     if (exception.error) {
```

Comment:
Worth noting that the code between the start of `collect` and this old guard clause ran in the world world, but doesn't run the new world (if the error collector is disable).  This includes a call to `_haveSeen`, which changes the state of the `ErrorCollector`.  However, the state changed by `_haveSeen` appears to only be used by `_haveSeen` itself, and no other methods in the agent call `_haveSeen` (hooray for respecting fake protected/private leading `_`) , so this should be fine.